psl-core/src/main/java/org/linqs/psl/evaluation/statistics/RankingEvaluator.java

```diff
@@ -0,0 +1,211 @@
+
+    // This is the truth value threshold that truth atoms must pass
+    // for us to use their corresponding target atom in a ranking
+    // metric calculation.
+    private double threshold;
+
+    // Save the map from target atoms to truth atoms
+    // for truth value lookups in ranking metric calculations.
+    private Map<RandomVariableAtom, ObservedAtom> labelMap;
+
+     // Maintain a map from predicted atoms
+     // to their ranks (with respect to their individual
+     // entities) for use in ranking metric calculations.
+    private Map<GroundAtom, Integer> rankMap;
+
+    public RankingEvaluator() {
+        this(RepresentativeMetric.valueOf(Options.EVAL_RANK_REPRESENTATIVE.getString()),
+                StringUtils.splitInt(Options.EVAL_RANK_CATEGORY_INDEXES.getString(), DELIM));
+    }
+
+    public RankingEvaluator(int... rawCategoryIndexes) {
+        this(Options.EVAL_RANK_REPRESENTATIVE.getString(), rawCategoryIndexes);
+    }
+
+    public RankingEvaluator(String representative, int... rawCategoryIndexes) {
+        this(RepresentativeMetric.valueOf(representative.toUpperCase()), rawCategoryIndexes);
+    }
+
+    public RankingEvaluator(RepresentativeMetric representative, int... rawCategoryIndexes) {
+        this.representative = representative;
+
+        categoryIndexes = new ArrayList<Integer>(rawCategoryIndexes.length);
+
+        for (int catIndex : rawCategoryIndexes) {
+            categoryIndexes.add(catIndex);
+        }
+
+        threshold = Options.EVAL_RANK_THRESHOLD.getDouble();
+        defaultPredicate = Options.EVAL_RANK_DEFAULT_PREDICATE.getString();
+    }
+
+    /**
+     * Grab the Constant arguments of an atom at specified positions.
+     */
+    private List<Constant> getArgsAtPositions(GroundAtom atom, List<Integer> indexes) {
+        Constant[] atomArgs = atom.getArguments();
+        List<Constant> argsAtPositions = new ArrayList<Constant>(indexes.size());
+        for (int index : indexes) {
+            argsAtPositions.add(atomArgs[index]);
+        }
+        return argsAtPositions;
+    }
+
+    @Override
+    public void compute(TrainingMap trainingMap) {
+        if (defaultPredicate == null) {
+            throw new UnsupportedOperationException("RankingEvaluators must have a default predicate set (through config).");
+        }
+
+        compute(trainingMap, StandardPredicate.get(defaultPredicate));
+    }
+
+    @Override
+    public void compute(TrainingMap trainingMap, StandardPredicate predicate) {
+        labelMap = trainingMap.getLabelMap();
+        rankMap = new HashMap<GroundAtom, Integer>();
+
+        int predicateArity = predicate.getArity();
+        List<Integer> entityIndexes = new ArrayList<Integer>(predicateArity - categoryIndexes.size());
+
+        for (int index = 0; index < predicateArity; index++) {
+            if (categoryIndexes.contains(index)) {
+                continue;
+            } else {
+                entityIndexes.add(index);
+            }
```

Comment:
Invert this guard and don't bother with the continue in this case
(so just a single `if` with no `else`).